api/settings_api.py

```diff
@@ -27,16 +32,26 @@ def do_post(self, **kwargs):
     user_pref = user_models.UserPref.get_signed_in_user_pref()
     if not user_pref:
       self.abort(403, msg='User must be signed in')
-    new_notify = self.get_bool_param('notify')
-    user_pref.notify_as_starrer = new_notify
+    raw_data = self.request.json
+    new_notify = raw_data.get('notify')
+
+    if not isinstance(new_notify, bool):
+        raise werkzeug.exceptions.BadRequest(
+            f"Expected boolean for 'notify', got {type(new_notify).__name__}"
+        )
+
+    request = PostSettingsRequest.from_dict(raw_data)
```

Comment:
We should probably only use the variable name `request` for the HTTPRequest object.  Instead this should be something like `settings_request` or `request_model`.